modules/host/contractmanager/writeaheadlog.go

```diff
@@ -318,9 +318,19 @@ func (wal *writeAheadLog) load() error {
 		}
 	})
 	ss := wal.cm.savedSettings()
-	err = persist.Save(settingsMetadata, ss, wal.fileSettingsTmp)
+	b, err := json.MarshalIndent(ss, "", "\t")
 	if err != nil {
-		build.ExtendErr("unable to write to settings temp file", err)
+		build.ExtendErr("unable to marshal settings data", err)
+	}
+	enc := json.NewEncoder(wal.fileSettingsTmp)
+	if err := enc.Encode(settingsMetadata.Header); err != nil {
+		build.ExtendErr("unable to write header to settings temp file", err)
+	}
+	if err := enc.Encode(settingsMetadata.Version); err != nil {
+		build.ExtendErr("unable to write version to settings temp file", err)
+	}
+	if _, err = wal.fileSettingsTmp.Write(b); err != nil {
+		build.ExtendErr("unable to write data settings temp file", err)
```

Comment:
not exactly sure what you are asking, but the primary difference is that the Sync isn't going to be called until the next iteration of the sync loop. That could be 500ms away. Sync returns a lot faster if the data is already written to disk, so we're actually writing it early to minimize the total amount of downtime (the contractmanager is stop-the-world while the syncs are happening - if we can do most of the disk writes ahead of time, it'll be stopped for less time).